x-pack/plugins/security_solution/public/flyout/document_details/right/components/related_alerts_by_same_source_event.tsx

```diff
@@ -31,24 +45,61 @@ export const RelatedAlertsBySameSourceEvent: React.VFC<RelatedAlertsBySameSource
   originalEventId,
   scopeId,
 }) => {
+  const { eventId, indexName, isPreviewMode } = useDocumentDetailsContext();
+  const { openLeftPanel } = useExpandableFlyoutApi();
+
   const { loading, dataCount } = useFetchRelatedAlertsBySameSourceEvent({
     originalEventId,
     scopeId,
   });
-  const text = (
-    <FormattedMessage
-      id="xpack.securitySolution.flyout.right.insights.correlations.sourceAlertsLabel"
-      defaultMessage="{count, plural, one {alert} other {alerts}} related by source event"
-      values={{ count: dataCount }}
-    />
+
+  const onClick = useCallback(() => {
+    openLeftPanel({
+      id: DocumentDetailsLeftPanelKey,
+      path: {
+        tab: LeftPanelInsightsTab,
+        subTab: CORRELATIONS_TAB_ID,
+      },
+      params: {
+        id: eventId,
+        indexName,
+        scopeId,
+      },
+    });
+  }, [eventId, indexName, openLeftPanel, scopeId]);
+
+  const text = useMemo(
```

Comment:
curious why the text and button were not added in `InsightSummaryRow`? The code looks almost identical in all 4 places

Reply:
that's because for prevalence we don't use the same text, we use a badge that just says `Uncomm`. I went back and forth with the idea of moving things into the `InsightSummaryRow` component but I didn't find a way I liked...

Here are some ideas I had:
- have a single `InsightSummaryRow` component that would - similar to how it was done until now - do all the scenarios we need, but then it would have to have some sort of if/else condition to support the clickable numbers and the `Uncommon` text for prevalence...
- then I thoughts about having the `InsightSummaryRow` component get a `text` and a `children` components but then it's nothing more than a `EuiFlexGroup` component... so no value added at all, even for unit testing
- one other option that wasn't too bad to my eyes was having 2 `InsightSummaryRow` components, one for the places where we show a number that's clickable (`Threat intelligence`, `Correlations`...), and one for `Prevalence` where we just show `Uncommon`. I was struggling to find a good component name though 😆 

If you have an idea I'm all ears!

Reply:
I see, the insight summary row is essentially a styled `euiFlexGroup` with 2 items. I think have 2 sets are not too bad, but it would be annoying if we have a third insight that is slightly different than these 2...

Reply:
@christineweng I've combined some of the logic in [this commit](https://github.com/elastic/kibana/pull/197349/commits/2f02ae2a2c033b67c7878aa9aaca51bafabdac4d), please take a look when you have some time. If you like this approach I will update the unit tests accordingly.

Here's what I decided to go with for now:
- move the click logic to the InsightsSummaryRow component, but I needed to pass the `subTab` via props because some actions go to the Threat Intelligence tab while others go to the Correlations tab
- I decided to have a single InsightsSummaryRow component because I really didn't like having 2 (and couldn't find 2 good names for them). The default behavior is that the `value` prop is receiving a `number`, which then shows the count as a `EuiBadge` and `EuiEmptyButton` with the click event handled internally. If the `value` prop is a `ReactElement` then we display that with no modifications\

I feel like this approach removes a lot of duplication, staying a bit opinionated while still providing some flexibility...
Let me know what you think!